### PR TITLE
Exit wptrunner with an error status if no tests ran

### PR DESCRIPTION
**Pull request by jgraham**
Typically no tests completing means that some error occured, or the user
supplied an invalid test path. In either case we should log the
problem and return a failure code.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/10030)

### CODE REVIEW

File: tools/wptrunner/wptrunner/wptrunner.py [hunk 1]
```diff
@@ -267,14 +269,18 @@ def run_tests(config, test_paths, product, **kwargs):
...
+                test_total += test_count
                 unexpected_total += unexpected_count
                 logger.info("Got %i unexpected results" % unexpected_count)
                 if repeat_until_unexpected and unexpected_total > 0:
                     break
                 logger.suite_end()
-    return unexpected_total == 0
+    if test_total == 0:
+        logger.error("No tests ran")
+    return test_total > 0 and unexpected_total == 0
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/wptrunner.py]: Nit:

```python
    if test_total == 0:
        logger.error("No tests ran")
        return False

    return unexpected_total == 0
```

File: tools/wptrunner/wptrunner/testrunner.py [hunk 2]
```diff
@@ -789,5 +793,11 @@ def stop(self):
         self.stop_flag.set()
         self.logger.debug("Stop flag set in ManagerGroup")
 
+    def test_count(self):
+        return sum(item.test_count for item in self.pool)
+
+    def subtest_count(self):
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/testrunner.py]: Unused?

**Review by Ms2ger — APPROVED** [review]

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-03-14 15:27:07
Finished: 2018-03-14 15:44:33


<h2>Failing Jobs</h2>
<ul>

<li>wptrunner_infrastructure</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25947)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/353378363)

**Comment by jgraham** [comment]
Of course landing this is hard until the underlying  issue from #9979 is fixed.

**Comment by gsnedders** [comment]
Should we just drop Chrome given #9979 so we can land this (and ensure Firefox doesn't regress), given that's not actually regressing testing Chrome? I'd be okay with that, and then #9979 becomes "re-enable Chrome" instead of "Chrome runs nothing".

File: testharness_runner.html [hunk 3]
```diff
@@ -1,6 +0,0 @@
-<!doctype html>
-<title></title>
-<script>
-var timeout_multiplier = 1;
-var win = null;
-</script>
```
Comment (gsnedders) [inline testharness_runner.html]: This seems unrelated to this commit.

File: tools/wptrunner/wptrunner/testrunner.py (hunk 2 above)
Comment (gsnedders) [reply tools/wptrunner/wptrunner/testrunner.py]: Yeah, this seems dead.

**Review by gsnedders — CHANGES_REQUESTED** [review]

**Comment by jgraham** [comment]
OK, I didn't mind the subtest count existing even though it's not yet used.

Reverted the inadvertent change.

**Review by gsnedders — APPROVED** [review]
